File: enc_dec.js
import CryptoJS from "crypto-js";

// keep secrets in .env file
var secret_key = import.meta.env.VITE_SECRET_KEY
var iv = import.meta.env.VITE_IV

//JS FIX IV ENCRYPTION CBC
export const encryptDataWithIv = (data) => {
  try {
		console.log("key len", AesConfig.SecretKey.length);
		var key = CryptoJS.enc.Utf8.parse(AesConfig.SecretKey);
		var iv = CryptoJS.enc.Utf8.parse(AesConfig.SecretIv);
		var encrypted = CryptoJS.AES.encrypt(data, key, {
		iv: iv,
		mode: CryptoJS.mode.CBC,
		});
		var encrypted1 = encrypted.toString();
		console.log("encryptDataWithIv: ", encrypted1);
		return encrypted1;
  } catch (err) {
    	console.log(`Error: ${err} please try again later`);
		return null
  }
};

//JS ENCRYPTION ECB
export const encryptData = (data) => {
  var key = CryptoJS.enc.Utf8.parse(AesConfig.SecretKey);
  var encrypted = CryptoJS.AES.encrypt(data, key, { mode: CryptoJS.mode.ECB });
  var encrypted2 = encrypted.toString();
  console.log("encryptData: ", encrypted2);
  return encrypted2;
};

export const decryptDataWithIv = (encryptedData) => {
  var iv = CryptoJS.enc.Utf8.parse(AesConfig.SecretIv);
  var key = CryptoJS.enc.Utf8.parse(AesConfig.SecretKey);
  var decrypted = CryptoJS.AES.decrypt(encryptedData, key, {
    iv: iv,
    mode: CryptoJS.mode.CBC,
  });
  // console.log("decrypted1", decrypted.toString(CryptoJS.enc.Utf8));
  return decrypted.toString(CryptoJS.enc.Utf8);
};

export const decryptData = (encryptedData) => {
  var key = CryptoJS.enc.Utf8.parse(AesConfig.SecretKey);
  var decrypted = CryptoJS.AES.decrypt(encryptedData, key, {
    mode: CryptoJS.mode.ECB,
  });
  console.log("decrypted2", decrypted.toString(CryptoJS.enc.Utf8));
  return decrypted.toString(CryptoJS.enc.Utf8);
};

File: enc_dec.py
import base64
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad,unpad

from dotenv import load_dotenv
import os

# Load environment variables from .env file
load_dotenv()
if load_dotenv():
    pass
else:
    print("env file is missing !!!!")


secret_key = os.getenv("SECRET_KEY") # configure it in .env file
iv = os.getenv("SECRET_IV")  # configure it in .env file

#AES ECB mode without IV

def encryptData(data):
        data = pad(data.encode(),16)
        cipher = AES.new(secret_key.encode('utf-8'), AES.MODE_ECB)
        return base64.b64encode(cipher.encrypt(data)).decode("utf-8", "ignore")

def decryptData(encrypted_data):
        encrypted_data = base64.b64decode(encrypted_data)
        cipher = AES.new(secret_key.encode('utf-8'), AES.MODE_ECB)
        return unpad(cipher.decrypt(encrypted_data),16).decode("utf-8", "ignore")



#AES CBC mode with IV

def encryptDataWithIv(data):
    Iv =  iv.encode('utf-8')
    data= pad(data.encode(),16)
    cipher = AES.new(secret_key.encode('utf-8'),AES.MODE_CBC,Iv)
    return base64.b64encode(cipher.encrypt(data)).decode("utf-8", "ignore")

def decryptDataWithIv(encrypted_data):   
    Iv =  iv.encode('utf-8')
    encrypted_data = base64.b64decode(encrypted_data)
    cipher = AES.new(secret_key.encode('utf-8'), AES.MODE_CBC, Iv)
    return unpad(cipher.decrypt(encrypted_data),16).decode("utf-8", "ignore")



'''
# message = 'Hi this is Encrypt Data Without Iv from Python'
# message2 = 'Hi this is Encrypt Data With Iv from python'

encrypted = encryptData(message)
print('encrypted ',encrypted)
 
decrypted = decryptData(encrypted)
print('data: ',decrypted)


print("\n\n")

encrypted = encryptDataWithIv(message2)
print('encrypted ',encrypted)

decrypted = decryptDataWithIv(encrypted)
print('data: ',decrypted)

'''
